src/main.py

```diff
@@ -233,26 +227,47 @@ def internet_on():
 		return False
 
 
-def gettoken():
-	token = mc.get("access_token")
-	refresh = config['alexa']['refresh_token']
-	if token:
-		return token
-	elif refresh:
+class Token(object):
+
+	_token = None
+	_timestamp = None
+	_validity = 3570
+
+	def __init__(self, aconfig):
+
+		self._aconfig = aconfig
+
+		self.renew()
+
+	def get(self):
+
+		if time.time() - self._timestamp > self._validity:
+			self.renew()
+
+		return self._token
+
+	def renew(self):
+
+		logger.debug("AVS token: Requesting a new one")
```

Comment:
And this should be `info`, this is good thing to see in log before enabling verbose logging.